openedx/core/djangoapps/content/search/api.py

```diff
@@ -467,6 +467,28 @@ def delete_index_doc(usage_key: UsageKey) -> None:
     _wait_for_meili_tasks(tasks)
 
 
+def delete_all_draft_docs_for_library(library_key: LibraryLocatorV2) -> None:
+    """
+    Deletes draft documents for the given XBlocks from the search index
+    """
+    current_rebuild_index_name = _get_running_rebuild_index_name()
+    client = _get_meilisearch_client()
+    # Delete all documents where last_published is null i.e. never published before.
+    delete_filter = [
+        f'{Fields.context_key}="{library_key}"',
+        # inner arrays are connected by an OR
+        [f'{Fields.last_published} IS EMPTY', f'{Fields.last_published} IS NULL'],
```

Comment:
@bradenmacdonald Meilisearch does not treat null values as empty. It only considers `"", [], {}` as empty. Currently, we don't really need to `IS EMTPY` as last_published is null if not published but I included it just in case if we are setting it to `""` somewhere. Do you suggest to remove it?

Reply:
I was just wondering if we really had both NULL and "" in the data, which would be kind of bad at this early stage, so I'm glad to hear it's consistent. Maybe just add a comment saying "This field should only be NULL or have a value, but we're also checking IS EMPTY just in case"

Reply:
Done.